### ISSUE TITLE
Allow opj_compress and opj_decompress to read/write images over stdin/stdout

### ISSUE THREAD

**Issue by gcode-importer** [uclouvain/openjpeg#224]
Originally reported on Google Code with ID 224

```
What steps will reproduce the problem?
1.piping openjepg with another applivcation BEFORE it
2.application sent picture to STDOUT
3.image_to_j2k do not recognize pipe input

What is the expected output? What do you see instead?
expected output is j2c image. Instead in case of "-" as input say unrecognized file,
without -i option say must have input image in PNG,TIFF.....

What version of the product are you using? On what operating system?
version 2.0 windows 7 64 bit

Please provide any additional information below.
convert image.tiff -(somefilter) rgb:- |image_to_j2k -i - -o output.j2c 
this pipe do mot work


```

Reported by `rado.markov` on 2013-06-08 15:35:45

**Comment by adolski** [comment]
+1 as this will enable the use of opj_decompress with Java's ProcessBuilder (and streams to/from arbitrary sources) without the hassle of JNI. Interestingly, doing something like this actually works, from the command line at least:

    ln -s /dev/stdin stdin.jp2
    ln -s /dev/stdout stdout.bmp
    opj_decompress -i stdin.jp2 -o stdout.bmp < source.jp2 > result.bmp

I've not been successful getting it to work with ProcessBuilder though.

**Comment by jronallo** [comment]
Yes, this would be a nice feature and, though not a performance gain for opj_decompress, would be for applications that use it.

**Comment by jpstroop** [comment]
If it helps in the meantime, I've used a named pipe when I've needed to do
this in the past.

**Comment by kamermans** [comment]
I see this issue was closed, along with its merged duplicate issues, and I see this in the changelog:

```
- Allow opj\_compress and opj\_decompress to read/write images over stdin/stdout [\#379](https://github.com/uclouvain/openjpeg/issues/379)
```

What I do not see, however, is any evidence in the code or documentation that this is possible.  I'm using   a version that I compiled today from `master` at commit level `66297f07a43d2770a97c8456d20202f3d051d980`.

Can someone please correct me if I'm wrong?  If I am wrong, and you can use STDIN (ex: `-i -:bmp`) and STDOUT (ex :`-o -:jp2`), then I'm happy to submit a PR with the updated documentation for `opj_compress` and `opj_decompress`.

Note: I do see code in `opj_decompress` (line 498, `infile_format()`) that attempts to identify the input file via magic numbers, which makes sense to me, so perhaps automatic type detection is only supported for this argument.
